### PR TITLE
Concave face Selection

### PR DESCRIPTION
**Pull request by AlaaCG**
Hi, 
the TexTools addon has overlapped selection, and I recently found out that Concave faces triggers overlapped selection after some frustration with why some islands were being selected when using overlapped. 

and most of the time one can't tell what's the reason or where are the concave faces and I didn't find a tool or an addon, so I thought it'd make sense for this to be a part of the TexTools addon, next to the overlapped. 

so I've added this button, tested it and it's working nicely.

the script for selecting concave was originally from here https://blender.stackexchange.com/questions/222153/how-to-select-concave-quads

### CODE REVIEW

**Comment by franMarz** [comment]
Thank you for the contribution. I'm busy right now, I hope I can look into this next week.

**Comment by franMarz** [comment]
Can you please share a blendfile with a geometry where Overlap fails due to a concave face? I can't replicate the issue and I'm hesitant to add this feature as is it now (I'd prefer to add it as an option of the Overlap operator that applies a different approach on concave faces if really needed)

**Comment by AlaaCG** [comment]
Hi again, Thanks for taking the time to look into this
I've attached 2 objects here, the ccv object has 4 concave faces in 4 islands, and the other one (fixed) where I deleted them all and re unwrapped

[concaves.zip](https://github.com/franMarz/TexTools-Blender/files/14469181/concaves.zip)

notice that in the ccv object, few islands are getting selected by the overlap operator, but it never selects the islands with concave faces in them

here is an example for the selected islands by the overlap operator, 
selected

and here are the islands with concave faces in them selected, 
concave islands

notice that only one island from the ones selected by overlap operator in the first image does contain 1 concave face, the overlap operator doesn't even select the islands with concave faces in them, and the fixed mesh (by merging the concave faces away) doesn't get any islands selected meaning that concaves trigger and confuse the overlap operator.

also found out later that blender select overlap sometimes does a better job than the overlap operator in tex tools, here is an example with the current mesh (although it selects different faces which have no problems), that also goes away when deleting the 4 concave faces 
blender overlap

neither blender's overlap or textools overlap operators are getting the actual islands with concave faces, and when deleting the concave faces both operators aren't selecting anything

I also had another problem last week where the textools overlap operator didn't select all overlapping islands and it missed an island that had vertices on top of an adjacent face, when trying with less geometry it selected it, so I don't know why or how it was missing that island, and on the other hand, blender's overlap selection was selecting the actual overlapping faces

so to summarize this, 
1- textools overlap operator doesn't work properly when there are concave faces in the mesh
2- textools overlap operator selects the island that contains overlapping geometry, while blender's overlap operator selects the exact overlapping faces which is far better, so I hope you would consider upgrading the overlap operator

**Comment by franMarz** [comment]
Thank you.

There is a problem with your example. There is no clear demonstration that concave faces are the cause of the error.

TexTools Overlap uses the Blender builtin "Select Overlap" operator, then extends the selection to the islands and deselects only one of them among all selected islands, randomly. This has its applications and I'm keeping the behaviour. When the vanilla Blender "Select Overlap" is better, just use it. Personally, I prefer the original function and use TexTools Overlap for quick tweaks.

The problem in your file is caused by self-overlapping, not overlapping between islands. This partially invalidates the functionality of TexTools Overlap. The self-overlapping in your mesh is caused by some degenerated UV faces. These faces are triangles whose vertices are aligned. They are present in the mesh as well. Blender 3D Space "Degenerate Dissolve" and TexTools "Select Degenerate" (Zero) operators are not clearing or detecting this for some reason, but anyway. Faces being concave is not the issue here.

I'll put this Pull Request on hold waiting for a clear example of concave faces being problematic with Blender builtin Select Overlap. Then, maybe we explore a workaround for that.

**Comment by AlaaCG** [comment]
I didn't know that textools uses blender's built in select overlap and tweaks it for more functionality, so I agree with you, my point was that blender's vanilla selects the faces only, but I think it's not a valid argument after what you clarified to me. 

thanks for explaining, 
I guess will have to wait until there is a clear example for it, if I'm able to produce one later, I will follow up here.

Have a good day

**Comment by franMarz** [comment]
I think the is_convex property is detecting aligned loops as not convex, but this is not the most reliable approach for finding degenerated geometry.
Next, I'll look into why Select Zero is not working properly in your mesh. And I'll make TexTools Overlap to not deselect the final self-overlapping island, so that particular issue doesn't need the use of the vanilla Select Overlap to be detected.

I'll close this Pull Request, please reopen it if you find something related with concave faces making Overlap fails.
